docs/code/uvwget/defs.h

```diff
@@ -0,0 +1,2 @@
+#define CONTAINER_OF(ptr, type, field)                                        \
+  ((type *) ((char *) (ptr) - ((char *) &((type *) 0)->field)))
```

Comment:
I'd probably use `offsetof()` from `<stddef.h>` here for legibility.  As well, can you add `#ifdef` include guards?  Thanks.